### PR TITLE
[5.1] Add multiple key support to the forget() method for Collections

### PR DESCRIPTION
**Pull request by blazeworx**
As discussed with @taylorotwell in Larachat/#internals, currently in `Collections` you can only `forget()` a single key: `$collection->forget('key');` this PR adds the ability to also specify an array of keys to forget, `$collection->forget(['foo', 'baz']);` without breaking if just a single string/key is provided.

Multiple Forget: 

``` php
$collection = new Collection(['name' => 'taylor', 'foo' => 'bar', 'baz' => 'qux']);
$collection->forget(['foo', 'baz']);
```

Results In:

```
>>> $collection->all();
=> [
       "name" => "taylor"
   ]
```

Single Forget (default 5.1):

``` php
$collection = new Collection(['name' => 'taylor', 'foo' => 'bar', 'baz' => 'qux']);
$collection->forget('foo');
```

Results In:

```
>>> $collection->all();
=> [
       "name" => "taylor",
       "baz"  => "qux"
   ]
```

### CODE REVIEW

**Comment by taylorotwell** [comment]
Does the doc block need to be adjusted?

**Comment by blazeworx** [comment]
@taylorotwell Oops, missed that one. I have updated the docblock accordingly.

File: src/Illuminate/Support/Collection.php [hunk 1]
```diff
@@ -232,12 +232,18 @@ public function flip()
...
      * Remove an item from the collection by key.
      *
-     * @param  mixed  $key
+     * @param  string|array  $key
      * @return $this
      */
     public function forget($key)
     {
-        $this->offsetUnset($key);
+        if (is_array($key)) {
```
Comment (lucasmichot) [inline src/Illuminate/Support/Collection.php]: Why not just casting the argument to an array? I would keep consistance with other functions acception either a string, either an array


        $keys = is_array($key) ? $key : [$key];

        foreach ($keys as $key) {
            $this->offsetUnset($key);
        }

        return $this;

Comment (GrahamCampbell) [inline src/Illuminate/Support/Collection.php]: What? Why not just:

```php
    foreach ((array) $keys as $key) {
        $this->offsetUnset($key);
    }

    return $this;
```

Comment (GrahamCampbell) [inline src/Illuminate/Support/Collection.php]: Most of laravel, is like not, `$keys = is_array($key) ? $key : [$key];`.

**Comment by blazeworx** [comment]
@GrahamCampbell :+1: I have updated it to your recommendation.

**Pull request merged**